src/color.cpp

```diff
@@ -6,756 +6,412 @@
-   case c_ltred_white  : return 69;
-   case c_green_white  : return 70;
-   case c_ltgreen_white  : return 71;
-   case c_brown_white  : return 72;
-   case c_yellow_white  : return 73;
-   case c_blue_white  : return 74;
-   case c_ltblue_white  : return 75;
-   case c_magenta_white  : return 76;
-   case c_pink_white  : return 77;
-   case c_cyan_white  : return 78;
-   case c_ltcyan_white  : return 79;
-
-   case c_black_green  : return 80;
-   case c_dkgray_green  : return 81;
-   case c_ltgray_green  : return 82;
-   case c_white_green  : return 83;
-   case c_red_green  : return 84;
-   case c_ltred_green  : return 85;
-   case c_green_green  : return 86;
-   case c_ltgreen_green  : return 87;
-   case c_brown_green  : return 88;
-   case c_yellow_green  : return 89;
-   case c_blue_green  : return 90;
-   case c_ltblue_green  : return 91;
-   case c_magenta_green  : return 92;
-   case c_pink_green  : return 93;
-   case c_cyan_green  : return 94;
-   case c_ltcyan_green  : return 95;
-
-   case c_black_yellow  : return 96;
-   case c_dkgray_yellow  : return 97;
-   case c_ltgray_yellow  : return 98;
-   case c_white_yellow  : return 99;
-   case c_red_yellow  : return 100;
-   case c_ltred_yellow  : return 101;
-   case c_green_yellow  : return 102;
-   case c_ltgreen_yellow  : return 103;
-   case c_brown_yellow  : return 104;
-   case c_yellow_yellow  : return 105;
-   case c_blue_yellow  : return 106;
-   case c_ltblue_yellow  : return 107;
-   case c_magenta_yellow  : return 108;
-   case c_pink_yellow  : return 109;
-   case c_cyan_yellow  : return 110;
-   case c_ltcyan_yellow  : return 111;
-
-   case c_black_magenta  : return 112;
-   case c_dkgray_magenta  : return 113;
-   case c_ltgray_magenta  : return 114;
-   case c_white_magenta  : return 115;
-   case c_red_magenta  : return 116;
-   case c_ltred_magenta  : return 117;
-   case c_green_magenta  : return 118;
-   case c_ltgreen_magenta  : return 119;
-   case c_brown_magenta  : return 120;
-   case c_yellow_magenta  : return 121;
-   case c_blue_magenta  : return 122;
-   case c_ltblue_magenta  : return 123;
-   case c_magenta_magenta  : return 124;
-   case c_pink_magenta  : return 125;
-   case c_cyan_magenta  : return 126;
-   case c_ltcyan_magenta  : return 127;
-
-   case c_black_cyan  : return 128;
-   case c_dkgray_cyan  : return 129;
-   case c_ltgray_cyan  : return 130;
-   case c_white_cyan  : return 131;
-   case c_red_cyan  : return 132;
-   case c_ltred_cyan  : return 133;
-   case c_green_cyan  : return 134;
-   case c_ltgreen_cyan  : return 135;
-   case c_brown_cyan  : return 136;
-   case c_yellow_cyan  : return 137;
-   case c_blue_cyan  : return 138;
-   case c_ltblue_cyan  : return 139;
-   case c_magenta_cyan  : return 140;
-   case c_pink_cyan  : return 141;
-   case c_cyan_cyan  : return 142;
-   case c_ltcyan_cyan  : return 143;
-
- }
- return 0;
+    for ( const auto& iter : mapColors ) {
+        if ( iter.second.first == color ) {
```

Comment:
At this point, nobody knows what `second`, `first` or what `second.first` (third?). is How about making a small struct that gives those values meaningful names:
```C++
struct invert_defs {
    std::string color_invert;
    std::string color_bright;
};
struct custom_defs {
    nc_color color_default;
    std::string color_custom;
};
```

You might know a better name, I'm not creative with naming.